zebra/ioctl.c

```diff
@@ -424,34 +421,81 @@ void if_get_flags(struct interface *ifp)
 			     safe_strerror(errno));
 		return;
 	}
-#ifdef HAVE_BSD_LINK_DETECT /* Detect BSD link-state at start-up */
 
-	/* Per-default, IFF_RUNNING is held high, unless link-detect says
-	 * otherwise - we abuse IFF_RUNNING inside zebra as a link-state flag,
-	 * following practice on Linux and Solaris kernels
+	if (!CHECK_FLAG(ifp->status, ZEBRA_INTERFACE_LINKDETECTION))
+		goto out;
+
+		/* Per-default, IFF_RUNNING is held high, unless link-detect
```

Comment:
this is indented oddly?

Reply:
I originally had it indented one tab left than it is now, but the  polychaeta  bot rejected it.
I would prefer it one tab left further left which I think is what you want, but how then is the bot overridden?

Reply:
I look at the suggestions and decide whether or not they make sense and if they don't I ignore them.

Reply:
ok